app/controllers/tweets_controller.rb

```diff
@@ -1,9 +1,36 @@
 class TweetsController < ApplicationController
 
-  def index
+before_action :authenticate_user!
+
+  def home
+    @tweet = Tweet.new
+    @tweets = Tweet.all
   end
 
   def new
+    @tweet = Tweet.new
+  end
+
+  def create
+    @user = current_user
+    @tweet = @user.tweets.new(tweet_params)
+    if @tweet.save
+      redirect_to home_path
+    else
+      @tweets = Tweet.all
```

Comment:
👍 The fact that you put this `@tweets` assignment here in the else fork shows you understand the use of instance variables!